app/docs/dev/policies/targetref.md

```diff
@@ -0,0 +1,333 @@
+
+Consider the example below:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshAccessLog
+metadata:
+  name: example
+  namespace: {{site.default_namespace}}
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef: # top level targetRef
+    kind: MeshService
+    name: web-frontend
+  to:
+    - targetRef: # to level targetRef
+        kind: MeshService
+        name: web-backend
+      default:
+        backends:
+          - file:
+              format:
+                plain: '{"start_time": "%START_TIME%"}'
+              path: "/tmp/logs.txt"
+  from:
+    - targetRef: # from level targetRef
+        kind: Mesh
+      default:
+        backends:
+          - file:
+              format:
+                plain: '{"start_time": "%START_TIME%"}'
+              path: "/tmp/logs.txt"
+```
+
+Using `spec.targetRef`, this policy targets all proxies that implement the service `web-frontend`.
+It defines the scope of this policy as applying to traffic either from or to `web-frontend` services.
+
+The `spec.to.targetRef` section enables logging for any traffic going to `web-backend`.
+The `spec.from.targetRef` section enables logging for any traffic coming from _any service_ in the `Mesh`.
+
+### Target resources
+
+Not every policy supports `to` and `from` levels. Additionally, not every resource can
+appear at every supported level. The specified top level resource can also affect which
+resources can appear in `to` or `from`.
+
+To help users, each policy documentation includes a table indicating which `targetRef` kinds is supported at each level.
+
+This table looks like:
+
+| `targetRef.kind`    | top level | to  | from |
+| ------------------- | --------- | --- | ---- |
+| `Mesh`              | ✅        | ✅  | ❌   |
+| `MeshSubset`        | ✅        | ❌  | ❌   |
+| `MeshService`       | ✅        | ❌  | ✅   |
+| `MeshServiceSubset` | ✅        | ❌  | ❌   |
+
+Here it indicates that the top level can use any targetRef kinds. But in `targetRef.to` only kind `Mesh` can be used and in `targetRef.from` only kind `MeshService`.
+
+
+### Merging configuration
+
+Because for a specific proxy multiple `targetRef` can be used.
+It is necessary to define a policy for merging configuration.
+
+We define a total order this way:
+
+- Mesh > MeshSubset > MeshService > MeshServiceSubset > MeshGatewayRoute (The more a `targetRef` is specific the higher priority it has)
+- If levels are equal the lexicographic order of policy names is used
+
+For `to` and `from` policies we concatenate the array for each matching policies.
+We then build configuration by merging each level using [JSON patch merge](https://www.rfc-editor.org/rfc/rfc7386).
+
+For example if I have 2 `default` ordered this way:
+
+```yaml
+conf: 1
+sub:
+  array: [1, 2, 3]
+  other: 50
+  other-array: [3, 4, 5]
+---
+sub:
+  array: []
+  other: null
+  other-array: [5, 6]
+  extra: 2
+```
+
+The merge result is:
+```yaml
+conf: 1
+sub:
+  array: []
+  other-array: [5, 6]
+  extra: 2
+```
```

Comment:
Can we have this prefixed with `spec` or some other way link this to the actual policy "world" so that the user is familiar with the constructs?